libpam/pam_handlers.c

```diff
@@ -936,10 +936,10 @@ int _pam_add_handler(pam_handle_t *pamh
 	} else {
 	    (*handler_p2)->argv = NULL;              /* no arguments */
 	}
-	if (((*handler_p2)->mod_name = extract_modulename(mod_path)) == NULL)
-	    return PAM_ABORT;
 	(*handler_p2)->grantor = 0;
 	(*handler_p2)->next = NULL;
+	if (((*handler_p2)->mod_name = extract_modulename(mod_path)) == NULL)
+	    return PAM_ABORT;
```

Comment:
This part of the change seems to be incomplete: there is another error path where these fields remain uninitialized.

Reply:
What do you think needs to be done with this part of the code?  I found it necessary to patch this too because this code is very similar to the one above.  (Although I haven't explored how to get to this part of the code)

Reply:
I'd just use calloc() for the *handler_p2 allocation so it is initialized with zeros.